tsconfig.all.jsonc

```
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "allowUnreachableCode": false, // Prevents accidental execution of unreachable code.
+    "allowUnusedLabels": false, // Disallows unused labels to avoid confusion in code.
+    "noFallthroughCasesInSwitch": true, // Ensures every switch case explicitly handles its execution flow.
+    "noUnusedLocals": true, // Reports unused local variables to keep the code clean.
+    "noErrorTruncation": true, // Prevents truncation of error messages for better debugging.
+    "isolatedModules": true, // Ensures each module can be transpiled independently for better tooling support.
+    "verbatimModuleSyntax": true, // Enforces explicit use of `type` imports for type-only dependencies.
+    "module": "nodenext", // Uses the ES module system compatible with Node.js.
+    "moduleResolution": "nodenext", // Resolves modules as per Node.js ESM standards.
+    "resolveJsonModule": true, // Allows importing JSON files as modules.
+    "types": ["vite/client", "vitest/globals", "node", "vitest"], // Provides Node.js global types and includes Vitest types.
```

We don't use `vite` and `node` types are already included by default

```suggestion
    "types": ["vitest/globals"],
```